src/main/java/com/ericsson/gerrit/plugins/eiffel/configuration/EiffelPluginConfiguration.java

```diff
@@ -67,12 +68,26 @@ public EiffelPluginConfiguration(final String pluginName, final NameKey project,
         }
         // Read plugin configuration
         this.enabled = pluginConfig.getBoolean(ENABLED, false);
-        this.filter = pluginConfig.getString(FILTER);
+        StringBuilder filterBuilder = new StringBuilder();
+        if (pluginConfig.getStringList(FILTER)!=null && pluginConfig.getStringList(FILTER).length > 0) {
+            Arrays.stream(pluginConfig.getStringList(FILTER)).forEach(filterval -> {
+                filterBuilder.append(filterval).append(",");
+            });
+        }
+        this.filter = filterBuilder.toString();
         this.remremPublishURL = pluginConfig.getString(REMREM_PUBLISH_URL);
         this.remremUsername = pluginConfig.getString(REMREM_USERNAME);
         this.remremPassword = pluginConfig.getString(REMREM_PASSWORD);
         // flow context is optional
-        this.flowContext = pluginConfig.getString(FLOW_CONTEXT);
+        StringBuilder flowContextBuilder = new StringBuilder();
+        if (pluginConfig.getStringList(FLOW_CONTEXT)!=null && pluginConfig.getStringList(FLOW_CONTEXT).length > 0) {
+            Arrays.stream(pluginConfig.getStringList(FLOW_CONTEXT)).forEach(filterval -> {
+                flowContextBuilder.append(filterval).append(",");
+            });
+        }
+        this.flowContext = flowContextBuilder.toString();
+        pluginConfig.setString(FILTER, filterBuilder.toString());
+        pluginConfig.setString(FLOW_CONTEXT, flowContextBuilder.toString());
```

Comment:
Why are we joining the []string value into a flat string and calling setString()? Don't we want to keep the filters and flow context strings as discrete string values in arrays or some other collection?

Reply:
The manual addition will override the configurations, so we need to set them so that they are visible in the GUI.